casper/src/test/scala/coop/rchain/casper/helper/TaskCasper.scala

```diff
@@ -0,0 +1,113 @@
+package coop.rchain.casper.helper
+
+import cats._
+import cats.effect.{ExitCase, Sync}
+import coop.rchain.blockstorage.LMDBBlockStore
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.comm.CommUtil.casperPacketHandler
+import coop.rchain.casper.util.comm.TransportLayerTestImpl
+import coop.rchain.casper.{
+  MultiParentCasper,
+  MultiParentCasperConstructor,
+  SafetyOracle,
+  ValidatorIdentity
+}
+import coop.rchain.catscontrib._
+import coop.rchain.comm._
+import coop.rchain.crypto.signatures.Ed25519
+import coop.rchain.metrics.Metrics
+import coop.rchain.p2p.EffectsTestInstances._
+import coop.rchain.p2p.effects.PacketHandler
+import coop.rchain.comm.connect.Connect.dispatch
+import coop.rchain.comm.protocol.routing._
+import coop.rchain.rholang.interpreter.Runtime
+import java.nio.file.Files
+
+import coop.rchain.casper.util.rholang.RuntimeManager
+import monix.execution.Scheduler
+import monix.eval.Task
+import scala.concurrent.duration.{FiniteDuration, MILLISECONDS}
+import scala.collection.mutable
+import coop.rchain.shared.PathOps.RichPath
+import scala.util.Random
+
+object TaskCasper {
+  def apply(sk: Array[Byte], genesis: BlockMessage)(
+      implicit scheduler: Scheduler): (Task[MultiParentCasper[Task]], () => Unit) = {
+    val blockStoreDir = BlockStoreTestFixture.dbDir
+    val runtimeDir    = BlockStoreTestFixture.dbDir
+
+    implicit val logEff           = new LogStub[Task]
+    implicit val timeEff          = new LogicalTime[Task]
+    implicit val errorHandlerEff  = ApplicativeError_.applicativeError[Task, CommError](appErrTask)
+    implicit val nodeDiscoveryEff = new NodeDiscoveryStub[Task]()
+    implicit val transportLayerEff =
+      new TransportLayerTestImpl[Task](HashSetCasperTestNode.peerNode("taskNode", 40400),
+                                       Map.empty[PeerNode, mutable.Queue[Protocol]])
+    implicit val metricEff = new Metrics.MetricsNOP[Task]
+    implicit val blockStoreEff =
+      LMDBBlockStore.create[Task](LMDBBlockStore.Config(blockStoreDir, 1024L * 1024))(syncInstance,
+                                                                                      metricEff)
+    implicit val turanOracleEffect = SafetyOracle.turanOracle[Task]
+
+    val activeRuntime  = Runtime.create(runtimeDir, 1024L * 1024)
+    val runtimeManager = RuntimeManager.fromRuntime(activeRuntime)
+    val validatorId    = ValidatorIdentity(Ed25519.toPublic(sk), sk, "ed25519")
+
+    val casperTask = for {
+      _        <- blockStoreEff.put(genesis.blockHash, genesis)
+      blockMap <- blockStoreEff.asMap()
+    } yield
+      MultiParentCasper.hashSetCasper[Task](runtimeManager, Some(validatorId), genesis, blockMap)
+
+    def cleanUp(): Unit = {
+      activeRuntime.close()
+      blockStoreEff.close()
+      runtimeDir.recursivelyDelete()
+      blockStoreDir.recursivelyDelete()
+    }
+
+    (casperTask, cleanUp _)
+  }
+
+  private val appErrTask = new ApplicativeError[Task, CommError] {
```

Comment:
I am not sure how this is gonna work. All the _error-handling_ type classes in `cats` (but also in `monix`, `fs2`) take the approach that allow only subclasses of `Throwable` to be errors. `CommError` is not one. Even here, you define `ApplicativeError[Task, CommError]` but `pure` is defined with `Throwable`.

Reply:
I don't really need it to work properly because throwing an exception in the tests is totally fine. I just need an instance of this to create `casper`. I don't know how @rabbitonweb magically creates the one for `Effect` in node, but we could ask him and try to follow a similar process.